### DISCUSSION TITLE
[help] Workflow for plots/images

### DISCUSSION THREAD

**Discussion by neilstats** [ropensci/targets#1100] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I am having trouble coming up with a satisfactory workflow in {targets} for generating images (i.e. plots) and then including them in an R markdown document. Maybe someone has a better suggestion or can correct any misunderstandings.

I can have functions that return ggplot2 or grid objects (or lists of such objects), then use `tar_load(plot1)` within an Rmd document. But I understand that saving such objects is not ideal (e.g. see this blog post: https://ropensci.org/blog/2022/12/06/save-ggplot2-targets/).

Instead, I can use a function that saves a plot to an image file and returns the file path, then use a file target and use something like `knitr::include_graphics(tar_read(plot1))` in my Rmd.

However, I would sometimes like a function to create multiple related plots (and treat them as one target). I realise a file target can track multiple files, so I can use `knitr::include_graphics(tar_read(plots)[1])` and `knitr::include_graphics(tar_read(plots)[2])` to include them in the Rmd.

But it would be nice to refer to the individual files by a name (e.g. `knitr::include_graphics(tar_read(plots)["plot1"])`), and file targets don't seem to keep the names of named vectors. I have tried using `tarchetypes::tar_files()`, and using a regular target for the named vector then a file target to track the files, like this:
```
tar_target(plots, make_figures()),
tar_target(plots_files, figures, format ="file")

## in the Rmd:
knitr::include_graphics(tar_read(plots)["plot1"])
```
but could not get the dependencies to be _quite_ correct. (I think there's a catch-22 situation with the plots and plots_files targets in this example.)

Any suggestions would be much appreciated! Many thanks.

**Comment by wlandau** [comment]
If `plots` is a file target which creates multiple plots in a directory:

```r
fs::dir_tree()
#> .
#> ├── plot_output_directory
#> │   ├── plot1.png
#> │   └── plot2.png
```

then `tar_read(plots)` will be:

```r
tar_read(plots)
#> [1] "plot_output_directory/plot1.png" "plot_output_directory/plot2.png"
```

In the report, you could assign informative names based on the base names of the file paths.

```r
plots <- tar_read(plots)
names(plots) <- fs::path_ext_remove(basename(plots)) # "plot1" "plot2"
knitr::include_graphics(tar_read(plots)["plot1"])
```
